### PR TITLE
feat(color): Add alternate layouts for model list in Manage Models page

### PR DESCRIPTION
**Pull request by philmoz**
Adds 4 layouts for the model list in the Manage Models page:
- Large image (2x2) - current layout and also default
- Small image (3x3)
- Name only, 2 columns (2x6)
- Name only, 1 column (1x6)

TODO:

- [ ] Companion support.

Button in the header changes layout - selection is saved to radio settings (YAML).

screenshot_tx16s_24-01-09_21-06-45
screenshot_tx16s_24-01-09_21-06-51
screenshot_tx16s_24-01-09_21-06-55
screenshot_tx16s_24-01-09_21-06-59

### CODE REVIEW

File: radio/src/datastructs_private.h [hunk 1]
```diff
@@ -956,6 +956,7 @@ PACK(struct RadioData {
 
   // Radio level tabs control (global settings)
 #if defined(COLORLCD)
+  uint8_t modelSelectLayout:2;
```
Comment (pfeerick) [inline radio/src/datastructs_private.h:959]: Any reason this shouldn't be NOBACKUP'd (and now that I look at it, most of the options afterwards)?

**Review by pfeerick — COMMENTED** [review]
Damn that was fast! It looks great! :)

**Comment by pfeerick** [comment]
And just while I'm thinking of it... I think there is a rare case where the sort option (bottom left) can is not set to a value... possibly on a new config / when it's never been set? i.e. I think a default value is missing. It was the case before when there were two buttons, but it's more obvious now as the field is blank.

**Comment by philmoz** [comment]
The default (0) was NO SORT - which got missed when the popup list was created for the sort order.
I've changed the code to force the Name ASC sort order if it is currently set to NO SORT.

Change the settings to NOBACKUP as well.

**Comment by robustini** [comment]
Top, i love it!

**Comment by rdeanchurch** [comment]
Excellent implementation.  Thank you.

**Comment by richardclli** [comment]
@philmoz Current category selection is multi select, so when I want to switch to another category, I need to tab another category to select that category and tab the original category to unselect the category.  Is this by design or did I missed something?

**Comment by pfeerick** [comment]
@richardclli Where have you been for last few months when it has come up in discussion, repeatedly? 🤣 

It was a by design (of the author) choice at the time the new model labels feature was added. I pointed out that behaviour, and wanted to have single and multi-select, but it was to hard to implement with the UI controls as they were at the time. Hard keys behave differently btw... `page >` / `page <` only selects one label at a time. 

There has been some discussion already in a few different issues on how to approach this... whether it be to display the content of both labels should the AND'ing of them fail to have any matches, whether the favourites label should AND, but all the others OR, etc, weither it should OR instead of AND, etc. My personal preference is it should be up to the user - as our own personal preferences don't necessarily match everyone's use case. As with the labels... not everyone wants to have the multi-select, but other people do. Same with these model list views... I'll probably never use them myself... I'm perfectly happy and prefer the "useless large model images" ... since it means I don't have to read the names to know which model I'm selecting, and it's a bigger target to poke on the screen :grin: But it seems some people want it. 🤷

**Comment by jtaylor2** [comment]
@richardclli , @pfeerick , @philmoz After the complaints on RCGroups and at my clubs about the way labels worked vs the former categories I looked into giving an option  between  multi select and single select.  I  didn't have enough understanding of the storage and translations areas to do a full pull request, but added a diff with some proof of concept code to  https://github.com/EdgeTX/edgetx/issues/4188 for comment.  It didn't receive any comments so I kind of dropped it.  I had hoped to receive comments on if  it was a good idea, had problems, or was the wrong way to go but didn't receive any.  The diff should still apply (just drop the .txt extension) if anybody wants to try it.

**Comment by pfeerick** [comment]
Was just a bad time of year I think... heading into December, Christmas and New Years. 

But that concept is one of the things I want to see added as a setting for that the model manager screen. i.e. maybe a settings cog button shown on the model select screen, then the option you proposed, the quick model select setting could be moved there and maybe a AND/OR option for labels would be the first candidates for an options screen.

**Comment by philmoz** [comment]
> @richardclli , @pfeerick , @philmoz After the complaints on RCGroups and at my clubs about the way labels worked vs the former categories I looked into giving an option between multi select and single select. I didn't have enough understanding of the storage and translations areas to do a full pull request, but added a diff with some proof of concept code to #4188 for comment. It didn't receive any comments so I kind of dropped it. I had hoped to receive comments on if it was a good idea, had problems, or was the wrong way to go but didn't receive any. The diff should still apply (just drop the .txt extension) if anybody wants to try it.

See PR #4532

**Comment by rotorman** [comment]
For feedback record - the users at RCGroups absolutely love this PR - see their comments from this post onwards one or two pages: https://www.rcgroups.com/forums/showthread.php?3916381-Official-EdgeTX-Discussion-Thread/page426#post51955753

**Comment by Ralf-W** [comment]
I tested this PR  - and I love it too. Well done!

There might however be a bug or issue:
When I flash my TX16S from this 2.10 test firmware version back to 2.9.2 some system settings have been changed: Especially my global function " Play Track 'engine off" / 'engine on' " when I flip switch SF has magically been changed from "play once" to "play every 1 second". 
I don't know yet if other setting have been changed automatically during flashing from 2.9.2 to 2.10(Beta), and back to 2.9.2.
My observation might be the hint to an unknown bug - or there is a simple explanation for the observed but unexpected misbehavior.

**Comment by pfeerick** [comment]
> I tested this PR - and I love it too. Well done!
> 
> There might however be a bug or issue: When I flash my TX16S from this 2.10 test firmware version back to 2.9.2 some system settings have been changed: Especially my global function " Play Track 'engine off" / 'engine on' " when I flip switch SF has magically been changed from "play once" to "play every 1 second". I don't know yet if other setting have been changed automatically during flashing from 2.9.2 to 2.10(Beta), and back to 2.9.2. My observation might be the hint to an unknown bug - or there is a simple explanation for the observed but unexpected misbehavior.

A simple explanation you flashed a newer firmware, which migrated settings to new format, and then flashed an older firmware, which doesn't like those values. If you are going to try newer firmware versions, you need to make a copy of your settings before updating - EdgeTX, just like OpenTX, is backward compatible and not forward compatible. You can user newer firmware with older settings, but not older firmware with newer settings. ;) Simplest way to protect yourself is swap microSD cards - which you can do on most of the radios - then there are no settings to get messed up ;) But still a good idea to have a backup of your RADIO / MODELS folder (or use Companion to save an ETX file) just in case anyway.

**Comment by ParkerEde** [comment]
@philmoz 
In the case of the 3x3 display, it would also be conceivable to make the text size of the model name one level smaller? With long text, the name completely covers the image. Alternatively, the text could also be trimmed so that there is no wrap if making the font smaller is not an option.
What do you think of an option to hide the name to show only the bitmaps in the case of 2x2 and 3x3?

**Comment by 3djc** [comment]
Hiding name would only work if bitmaps are unique enough to recognise model, which is, more often than not, not the case.

**Comment by ParkerEde** [comment]
This is exactly why I thought of an option switch so that those for whom it is clearly maintained could hide the text if they wish.

**Comment by philmoz** [comment]
> In the case of the 3x3 display, it would also be conceivable to make the text size of the model name one level smaller? With long text, the name completely covers the image. Alternatively, the text could also be trimmed so that there is no wrap if making the font smaller is not an option. What do you think of an option to hide the name to show only the bitmaps in the case of 2x2 and 3x3?

Good suggestions, thanks.
I've made the following changes:
- Default font size for 3x3 display is XS (one size smaller)
- If text does not fit available width, reduce font size (STD -> XS -> XXS)
- Don't wrap long lines.

**Comment by ParkerEde** [comment]
Yes, fine. I have tested it. Looks good. Thank you very much

**Comment by pfeerick** [comment]
Looks great on TX16, still not sure I'd use it myself as I have images for all of my models (either stock or taken myself) and like the full sized icons... but options are always good :)

**Pull request merged**